src/ui/common/src/components/workflows/artifact/metric/history.tsx

```diff
@@ -62,10 +62,18 @@ const MetricsHistory: React.FC<Props> = ({ historyWithLoadingStatus }) => {
       }
     ),
   };
-
-  const dataToPlot = historicalData.data.filter(
-    (x) => !!x['timestamp'] && !!x['value']
+  const dataSortedByLatest = historicalData.data.sort(
+    (x, y) =>
+      Date.parse(y['timestamp'] as string) -
```

Comment:
What do y['timestamp'] and x['timestamp'] look like? Ideally it's just time in milliseconds since the unix epoch right? If so, we don't need to parse these dates out.

Reply:
I believe you could do something like x.getTime() and y.getTime() to get a number back that you can use for comparison:
<img width="246" alt="image" src="https://user-images.githubusercontent.com/1031759/205989986-7b1150f6-a9f9-4759-a460-0ca6bfbde68a.png">

Reply:
<img width="653" alt="Screen Shot 2022-12-06 at 11 24 07 AM" src="https://user-images.githubusercontent.com/78303449/206003452-f97105c3-0170-43c1-9fd0-9b9e507acd1f.png">
@agiron123 The timestamp here is the Datetime object converted to string I believe, which is why I converted back to Date. One way we can do it is simply store the timestamp as `Date` object. Is there any reason we cast it to a string over here?

Reply:
we convert to a string so that it shows nicely like 1/1/2023 4:00pm